src/parquet/util/crypto.cc

```diff
@@ -0,0 +1,324 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+
+#include <openssl/aes.h>
+#include <openssl/evp.h>
+#include <openssl/rand.h>
+#include <string>
+#include <sstream>
+#include <iostream>
+#include "parquet/exception.h"
+#include "parquet/types.h" 
+
+#include "parquet/util/crypto.h"
```

Comment:
I have a build issue on Windows in types.h (line 93), it recognize OPTIONAL as something else rather than an enum. When I include openssl/*.h files below line 28 in crypto.cc, it works. It seems that there is some macro on openssl related to OPTIONAL. So could you please change the order of include here?